numba/testing.py

```diff
@@ -1,9 +1,522 @@
+                return _make_skipped_test(name, e, self.suiteClass)
+            else:
+                return super(TestLoader, self).discover(start_dir, pattern, top)
+        else:
+            try:
+                __import__(start_dir)
+                module = sys.modules[start_dir]
+                load_tests = getattr(module, 'load_tests', None)
+                if load_tests:
+                    return load_tests(self, [], pattern)
+            finally:
+                return super(TestLoader, self).discover(start_dir, pattern, top)
+            
+
+
+# "unittest.main" is really the TestProgram class!
+# (defined in a module named itself "unittest.main"...)
+
+class NumbaTestProgram(unittest.main):
+    """
+    A TestProgram subclass adding the following options:
+    * a -R option to enable reference leak detection
+    * a --profile option to enable profiling of the test run
+    * a -m option for parallel execution
+    * a -l option to (only) list tests
+
+    Currently the options are only added in 3.4+.
+    """
+
+    refleak = False
+    profile = False
+    multiprocess = False
+    list = False
+    
+    def __init__(self, *args, **kwargs):
+        # Disable interpreter fallback if we are running the test suite
+        if config.COMPATIBILITY_MODE:
+            warnings.warn("Unset INTERPRETER_FALLBACK")
+            config.COMPATIBILITY_MODE = False
+
+        # The default test loader is buggy in its handling of load_tests().
+        # See http://bugs.python.org/issue25520
+        kwargs['testLoader'] = TestLoader()
+
+        # HACK to force unittest not to change warning display options
+        # (so that NumbaWarnings don't appear all over the place)
+        sys.warnoptions.append(':x')
+        self.nomultiproc = kwargs.pop('nomultiproc', False)
+        super(NumbaTestProgram, self).__init__(*args, **kwargs)
+
+    def _getParentArgParser(self):
+        # NOTE: this hook only exists on Python 3.4+. The options won't be
+        # added in earlier versions (which use optparse - 3.3 - or getopt()
+        # - 2.x).
+        parser = super(NumbaTestProgram, self)._getParentArgParser()
+        if self.testRunner is None:
+            parser.add_argument('-R', '--refleak', dest='refleak',
+                                action='store_true',
+                                help='Detect reference / memory leaks')
+        parser.add_argument('-m', '--multiprocess', dest='multiprocess',
+                            action='store_true',
+                            help='Parallelize tests')
+        parser.add_argument('-l', '--list', dest='list',
+                            action='store_true',
+                            help='List tests without running them')
+        parser.add_argument('--profile', dest='profile',
+                            action='store_true',
+                            help='Profile the test run')
+        return parser
+
+    def parseArgs(self, argv):
+        if '-l' in argv:
```

Comment:
I'm not sure I understand the purpose. There isn't anything that's available in recent (or even future) Python versions, or is there ?